packages/peregrine/lib/talons/Gallery/useGalleryItem.js

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from 'react';
+import { useLazyQuery } from '@apollo/react-hooks';
+
+/**
+ *
+ * @param {Object} props
+ * @param {Object} props.item - the product item being rendered
+ * @param {Object} props.queries - the queries for the gallery item
+ */
+export const useGalleryItem = props => {
+    const { item, queries } = props;
+
+    const { prefetchProductQuery } = queries;
+    const { url_key } = item;
+
+    // Prefetch the underlying product for this gallery item.
+    const [runPrefetchQuery] = useLazyQuery(prefetchProductQuery);
```

Comment:
💯 

I like this approach because it re-uses the `GET_PRODUCT_DETAIL_QUERY` and doesn't delay the rendering of the `GalleryItem`.

Another thought is to have the Service Worker do it, but then we have to keep those GraphQL fragments / queries in sync and we don't have a system for that right now.

Reply:
Solved through use of IntersectionObserver